internal/pkg/warmup/warmup.go

```diff
@@ -132,7 +134,11 @@ func (w Warmup) HTTPWarmupWorker(wg *sync.WaitGroup, requests <-chan http.Reques
 	for request := range requests {
 		time.Sleep(time.Duration(requestDelayMilliseconds) * time.Millisecond)
 
-		resp := w.Target.httpClient.SendRequest(request.Method, request.Path, headers, request.Body)
+		headersMap := util.ToHeaders(headers)
+		// Overwrite Content-Encoding header if required
+		maps.Copy(headersMap, request.Headers)
```

Comment:
If the request has a different value for the `Content-Encoding` header this will overwrite it. I am assuming this is ok given that anyone using mittens is also the owner of the service.

Reply:
Going through the flow again, if a value is provided by the user for this header this choice will be respected based on the signature `(maps.Copy(dst, src))`. Hope the suggestion in the comment makes it clear.
PS: I think we need more coverage in terms of tests for the headers functionality but this could be part of a separate PR.

Reply:
https://pkg.go.dev/golang.org/x/exp/maps#Copy

dst is 1st param
src is 2nd param

Hence a `Content-Encoding` header value provided by a user is overwritten. But yes, there should be a test for it.

Reply:
You are right. `headersMap` are the user-set headers and `request.Headers` the ones we set during the execution.
Names are a bit confusing, we will refactor it.